index.js

```diff
@@ -153,134 +52,81 @@ function getSha(imageName) {
-    repo: "notification-manifests",
-    branch: branchName,
-    sha: releaseConfig.sha,
-    path: "env/production/kustomization.yaml",
-    message: `Updated manifests to ${manifestUpdates}`,
-    content: newReleaseContentBlob,
-  });
-
-  const pr = await octokit.pulls.create({
-    owner: GH_CDS,
-    repo: "notification-manifests",
-    title: `[AUTO-PR] Automatically generated new release ${new Date().toISOString()}`,
-    head: branchName,
-    base: "main",
-    body: issueContent.replace(
-      "> Give details ex. Security patching, content update, more API pods etc",
-      logs
-    ),
-    draft: true,
-  });
-  return Promise.all([ref, update, pr]);
-}
-
-async function hydrateWithSHAs(releaseConfig, projects) {
+async function hydrateWithSHAs() {
   return await Promise.all(
-    releaseConfig.images.map(async (image) => {
-      const matchingProject = projects.find((project) =>
-        image.newName.includes(project.ecrName)
+    PROJECTS.map(async (project) => {
+      project.headSha = await getHeadSha(project.repoName);
+      project.shortSha = shortSha(project.headSha)
+      project.headUrl = getLatestImageUrl(
+        PROJECTS,
+        project.ecrName,
+        project.headSha
       );
-      matchingProject.headSha = await getHeadSha(matchingProject.name);
-      matchingProject.headUrl = getLatestImageUrl(
-        matchingProject.ecrName,
-        matchingProject.headSha
+
+      const releaseContent = await getContents(
+        "notification-manifests",
+        project.manifestFile
       );
-      matchingProject.oldSha = getSha(image.newName);
-      matchingProject.oldUrl = image.newName;
-      image.newName = matchingProject.headUrl;
-      return matchingProject;
+
+      const originalFileContents = Base64.decode(releaseContent.content)
+      const re = new RegExp(`${project.ecrName}:\\S*`, "g");
+      project.oldUrl = originalFileContents.match(re)[0]
+      project.oldSha = getSha(project.oldUrl);
+      return project;
     })
   );
 }
 
+function UpdateContents(content, project) {
+  return content.replace(`${project.ecrName}:${project.oldSha}`, `${project.ecrName}:${shortSha(project.headSha)}`)
+}
+
 // Main ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 async function run() {
-  const releaseContent = await getContents(
-    GH_CDS,
-    "notification-manifests",
-    "env/production/kustomization.yaml"
-  );
-
   const prTemplate = await getContents(
-    GH_CDS,
     "notification-manifests",
     ".github/PULL_REQUEST_TEMPLATE.md"
   );
-
-  const releaseConfig = YAML.parse(Base64.decode(releaseContent.content));
   const issueContent = Base64.decode(prTemplate.content);
+  const manifestFiles = Array.from(new Set(PROJECTS.map(project => project.manifestFile)))
+
+  await hydrateWithSHAs();
+
+  var changesToManifestFiles = manifestFiles.map(async (manifestFile) => {
+    const projects = PROJECTS.filter(project => project.manifestFile == manifestFile)
```

Comment:
brainstorm: If there was a zip operation to build a set of manifest files with their associated project, the filter would not be necessary. This is where we could benefit with using a functional library such as lodash to make it a bit clearer and terse.

Reply:
We can use a reducer, for example
```
const reducer = (previous, project) => ({ ...previous, [project.manifestFile]: (previous[project.manifestFile] || []).concat(project) })
const projectsForFiles = projects.reduce(reducer, {})
```